File: MonolithFullStackTemplate/templateUI/src/app/people/person-photo/person-photo.ts
export class PersonPhoto {
  constructor(
    public id: number,
    public fileId: number,
    public name: string,
    public mimeType: string,
    public personId: string,
    public size: number,
    public links: {
      self: {
        href: string
      },
      downloadPersonPhoto?: {
        href: string
      }
    }
  ) {  }
}

File: MonolithFullStackTemplate/templateAPI/src/main/java/com/template/people/images/PersonPhoto.java
package com.template.people.images;


import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.io.Serializable;

@Entity
@Table(name = "tp_person_photo")

@Getter
@Setter
@ToString
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class PersonPhoto implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    private Long personId;

    @NotNull
    private String fileId;

    public void updateFileId(String fileId) {
        this.fileId = fileId;
    }
}
